plugin/cache.go

```diff
@@ -87,6 +88,10 @@ type opFunc func() (interface{}, error)
 // For example, CachedOp could be useful to cache an API request whose response
 // lets you implement Attr() and Metadata() for the given entry.
 func CachedOp(opName string, entry Entry, ttl time.Duration, op opFunc) (interface{}, error) {
+	if !opNameRegex.MatchString(opName) {
```

Comment:
This isn't quite right. `opNameRegex` allows anything to follow after the regex, it needs to end with `$`.